### PR TITLE
Experimental scanner improvements

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Please see the commit messages for details.

### CODE REVIEW

File: scanner/src/main/kotlin/experimental/ScanController.kt [hunk 1]
```diff
@@ -140,7 +140,7 @@ class ScanController(
      * Get the [NestedProvenance] for the provided [id], or null if no nested provenance for the [id] is available.
      */
     fun getNestedProvenance(id: Identifier): NestedProvenance? =
-        nestedProvenances[packageProvenancesWithoutVcsPath.getValue(id)]
+        nestedProvenances[packageProvenancesWithoutVcsPath[id]]
```
Comment (sschuberth) [inline scanner/src/main/kotlin/experimental/ScanController.kt:143]: Hmm, your intended "Instead return null" seems to be achieved like this: `packageProvenancesWithoutVcsPath[id]` returns null, then that null value is being looked in `nestedProvenances[packageProvenancesWithoutVcsPath`, which fails and finally also returns null. Is this "indirection" intended? And / or should this maybe be written for clarify like

    packageProvenancesWithoutVcsPath[id]?.let { nestedProvenances[it] }

Comment (mnonnenmacher) [reply scanner/src/main/kotlin/experimental/ScanController.kt:143]: I thought that's just the simplest way to achieve that without adding more code. I'm fine with changing it to the suggestion as well, if that's what you prefer?

Comment (sschuberth) [reply scanner/src/main/kotlin/experimental/ScanController.kt:143]: I'm fine with the code, but I'd appreciate some explanation in the commit message about this "indirection". Or keep the commit message and change the code 😉 But I'll approve anyway and leave it to you.

**Review by sschuberth — APPROVED** [review]

File: scanner/src/main/kotlin/experimental/ScanController.kt (hunk 1 above)
Comment (mnonnenmacher) [reply scanner/src/main/kotlin/experimental/ScanController.kt]: Changed as requested.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**